pkg/streamingpromql/engine_test.go

```diff
@@ -1215,3 +1215,88 @@ func (t *timeoutTestingQueryTracker) Insert(ctx context.Context, _ string) (int,
 func (t *timeoutTestingQueryTracker) Delete(_ int) {
 	panic("should not be called")
 }
+
+func TestAnnotations(t *testing.T) {
+	startT := timestamp.Time(0).Add(time.Minute)
+	step := time.Minute
+	endT := startT.Add(2 * step)
+
+	mixedFloatHistogramData := `
+		metric{type="float", series="1"} 0+1x3
+		metric{type="float", series="2"} 1+1x3
+		metric{type="histogram", series="1"} {{schema:0 sum:0 count:0}}+{{schema:0 sum:5 count:4 buckets:[1 2 1]}}x3
+		metric{type="histogram", series="2"} {{schema:0 sum:1 count:1 buckets:[1]}}+{{schema:0 sum:5 count:4 buckets:[1 2 1]}}x3
+	`
+
+	testCases := map[string]struct {
+		data                       string
+		expr                       string
+		expectedWarningAnnotations []string
+		expectedInfoAnnotations    []string
```

Comment:
No test cases for Info annotations?

Also a test case for multiple annotations would also be good to ensure we are appending correctly.

Reply:
>No test cases for Info annotations?

Not yet - we're not yet emitting any info annotations. That will come in a separate PR (eg. the various annotations for `rate`).

>Also a test case for multiple annotations would also be good to ensure we are appending correctly.

Because we only support a single kind of annotation, and that annotation always has the same message, there's no way to construct a test case for this yet.

For example, the expression `sum by (series) (metric) + sum by (series) (metric)` only produces a single annotation even though both `sum`s would emit a warning about a mix of floats and histograms. This is because the deduplication that `annotations.Annotations` does does not consider the position attached to the annotation, and the message is the same in both cases. (Both Prometheus' engine and Mimir's engine are consistent in this regard.)